fluent-bundle/src/scope.ts

```diff
@@ -34,10 +34,27 @@ export class Scope {
     this.errors.push(error);
   }
 
-  memoizeIntlObject<ObjectT extends object, OptionsT>(
-    ctor: new (locales: Array<string>, opts: OptionsT) => ObjectT,
-    opts: OptionsT
-  ): ObjectT {
+  memoizeIntlObject(
+    ctor: typeof Intl.NumberFormat,
+    opts: Intl.NumberFormatOptions
+  ): Intl.NumberFormat;
+  memoizeIntlObject(
+    ctor: typeof Intl.DateTimeFormat,
+    opts: Intl.DateTimeFormatOptions
+  ): Intl.DateTimeFormat;
+  memoizeIntlObject(
+    ctor: typeof Intl.PluralRules,
+    opts: Intl.PluralRulesOptions
+  ): Intl.PluralRules;
+  memoizeIntlObject(
+    ctor:
+      | typeof Intl.NumberFormat
+      | typeof Intl.DateTimeFormat
+      | typeof Intl.PluralRules,
+    opts: Intl.NumberFormatOptions &
+      Intl.DateTimeFormatOptions &
+      Intl.PluralRulesOptions
+  ): Intl.NumberFormat | Intl.DateTimeFormat | Intl.PluralRules {
```

Comment:
While this function used to be parametric, it should be easier to use with the overloads that allow the return type to be correctly typed.

Reply:
Question: How does typescript refine the usage here? Will the return type always be `Intl.NumberFormat | Intl.DateTimeFormat | Intl.PluralRules`, or can it refine the union to a single `Intl.NumberFormat`? I'm a bit surprised by this change, but perhaps this was a new warning.

Reply:
The immediately preceding [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) allow for the return type to match the argument types.

And yes, there was a new linter error about the preceding `object` usage.